src/Chapter3.hs

```diff
@@ -970,7 +1104,31 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
-
+data WeekDays
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Show, Read, Eq, Ord, Enum)
+
+isWeekend :: WeekDays -> Bool
+isWeekend d = d == Saturday || d == Sunday
+
+nextDay :: WeekDays -> WeekDays
+nextDay d
+    | d == Sunday = Monday
+    | otherwise = toEnum ((fromEnum d) + 1)
+
+daysToParty :: WeekDays -> Int
+daysToParty d
+    | d >= Friday = 7 - dayNum + daysToParty Monday
+    | otherwise = fromEnum Friday - dayNum
+      where
+        dayNum :: Int
+        dayNum = fromEnum d
```

Comment:
Your implementation is absolutely correct :+1: 
Though, you can use knowledge from Chapter One on how `mod` works for negative numbers to implement this function in a more elegant way